wire-compiler/src/main/java/com/squareup/wire/schema/Root.kt

```diff
@@ -59,8 +63,9 @@ internal fun Location.roots(
     }
     throw IllegalArgumentException("unable to resolve $this")
   } else {
-    val path = fs.getPath(path)
-    return path.roots(closer, this)
+    return baseToRoots.computeIfAbsent(path) {
+      fs.getPath(this.path).roots(closer, this)
```

Comment:
nit: keep the temp `val path` for readability

Reply:
done